lib/core/viewer/components/viewer.component.ts

```diff
@@ -720,4 +768,86 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
     private generateCacheBusterNumber() {
         this.cacheBusterNumber = Date.now();
     }
+
+    private async displayNodeVersionRendition(nodeId: string, versionId: string) {
+        try {
+            const rendition = await this.resolveVersionRendition(nodeId, versionId, 'pdf');
+            if (rendition) {
+                const renditionId = rendition.entry.id;
+
+                if (renditionId === 'pdf') {
+                    this.viewerType = 'pdf';
+                } else if (renditionId === 'imgpreview') {
+                    this.viewerType = 'image';
+                }
+
+                this.urlFileContent = this.apiService.contentApi.getVersionRenditionUrl(nodeId, versionId, renditionId);
+            }
+        } catch (err) {
+            this.logService.error(err);
+        }
+    }
+
+    private async resolveVersionRendition(nodeId: string, versionId: string, renditionId: string): Promise<RenditionEntry> {
```

Comment:
this code seems quite duplicated of the normal version retention I think we should try to unify the common logic and move it in a service level